src/Aspire.Hosting.Keycloak/KeycloakResouceBuilderExtensions.cs

```diff
@@ -0,0 +1,129 @@
+/// Provides extension methods for adding Keycloak resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class KeycloakResouceBuilderExtensions
+{
+    private const string AdminEnvVarName = "KEYCLOAK_ADMIN";
+    private const string AdminPasswordEnvVarName = "KEYCLOAK_ADMIN_PASSWORD";
+    private const int DefaultContainerPort = 8080;
+    private const string RealmImportDirectory = "/opt/keycloak/data/import";
+
+    /// <summary>
+    /// Adds a Keycloak container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. </param>
+    /// <param name="port">The host port that the underlying container is bound to when running locally.</param>
+    /// <param name="adminUsername">The parameter used as the admin for the Keycloak resource. If <see langword="null"/> a default value will be used.</param>
+    /// <param name="adminPassword">The parameter used as the admin password for the Keycloak resource. If <see langword="null"/> a default password will be used.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// The container is based on the quay.io/keycloak/keycloak container image.
+    /// The default tag is 24.0. The container exposes port 8080 by default.
+    /// </remarks>
+    /// <example>
+    /// <code lang="csharp">
+    /// var keycloak = builder.AddKeycloak("keycloak");
+    ///
+    /// var myService = builder.AddProject&lt;Projects.MyService&lt;()
+    ///                        .WithReference(keycloak);
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<KeycloakResource> AddKeycloak(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? adminUsername = null,
+        IResourceBuilder<ParameterResource>? adminPassword = null)
+    {
+        var passwordParameter = adminPassword?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
+
+        var resource = new KeycloakResource(name, adminUsername?.Resource, passwordParameter);
+
+        var keycloak = builder
+            .AddResource(resource)
+            .WithImage(KeycloakContainerImageTags.Image)
+            .WithImageRegistry(KeycloakContainerImageTags.Registry)
+            .WithImageTag(KeycloakContainerImageTags.Tag)
+            .WithHttpEndpoint(port: port, targetPort: DefaultContainerPort)
+            .WithEnvironment(context =>
+            {
+                context.EnvironmentVariables[AdminEnvVarName] = resource.AdminReference;
+                context.EnvironmentVariables[AdminPasswordEnvVarName] = resource.AdminPasswordParameter;
+            });
+
+        if (builder.ExecutionContext.IsRunMode)
+        {
+            keycloak.WithArgs("start-dev");
```

Comment:
What does `start-dev` do?

I'm wondering if this should be tied to the `"Development"` environment instead of "IsRunMode".

Reply:
From Keycloak docs:

This mode offers convenient defaults for developers, such as for developing a new Keycloak theme.

Development mode sets the following default configuration:
* HTTP is enabled
* Strict hostname resolution is disabled
* Cache is set to local (No distributed cache mechanism used for high availability)
* Theme-caching and template-caching is disabled

I thought "IsRunMode" is effectively true when folks run their Aspire solution localy?

Reply:
I think `IsRunMode` is correct. For the most part we aren't using the .NET environment is Aspire - we leave that free for developers to change things themselves.